webui/react/src/e2e/tests/projectsWorkspaces.spec.ts

```diff
@@ -239,32 +247,143 @@ test.describe('Workspace UI CRUD', () => {
+    const workspaceList = new WorkspaceList(authedPage);
+
+    const namesAfterNewest = await getCurrentWorkspaceNames(workspaceList);
+    const idSortedWorkspaceNames = _.orderBy(workspaces, 'id', 'desc').map((w) => w.name);
+    expect(idSortedWorkspaceNames).toEqual(
+      namesAfterNewest.filter((n) => {
+        return n && workspaces.map((w) => w.name).includes(n);
+      }),
+    );
+
+    await workspaceList.sortSelect.selectMenuOption('Alphabetical');
+
+    const namesAfterAlphabetical = await getCurrentWorkspaceNames(workspaceList);
+    const nameSortedWorkspaceNames = _.orderBy(workspaces, 'name', 'asc').map((w) => w.name);
+    expect(nameSortedWorkspaceNames).toEqual(
+      namesAfterAlphabetical.filter((n) => {
+        return n && workspaces.map((w) => w.name).includes(n);
+      }),
+    );
+  });
+
+  test('Filter', async ({ authedPage }) => {
+    const workspaceList = new WorkspaceList(authedPage);
+
+    await workspaceList.newWorkspaceButton.pwLocator.click();
+    const currentUserWorkspaceName = await createWorkspaceAllFields(
+      workspaceList.createModal,
+      'fromButton',
+    );
+    await workspaceList.nav.sidebar.workspaces.pwLocator.click();
+
+    const otherUserWorkspaceName = workspaces.map((w) => w.name)[0];
+    const namesAfterAll = await getCurrentWorkspaceNames(workspaceList);
+    expect(namesAfterAll).toContain(otherUserWorkspaceName);
+    expect(namesAfterAll).toContain(currentUserWorkspaceName);
+
+    await workspaceList.whoseSelect.selectMenuOption("Others' Workspaces");
+    const namesAfterOthers = await getCurrentWorkspaceNames(workspaceList);
+    expect(namesAfterOthers).toContain(otherUserWorkspaceName);
+    expect(namesAfterOthers).not.toContain(currentUserWorkspaceName);
+
+    await workspaceList.whoseSelect.selectMenuOption('My Workspaces');
+    const namesAfterMy = await getCurrentWorkspaceNames(workspaceList);
+    expect(namesAfterMy).toContain(currentUserWorkspaceName);
+    expect(namesAfterMy).not.toContain(otherUserWorkspaceName);
+
+    // cleanup:
+    const deleteModal = workspaceList.deleteModal;
+    const workspaceCard = workspaceList.cardByName(currentUserWorkspaceName);
+    await (await workspaceCard.actionMenu.open()).delete.pwLocator.click();
+    await deleteModal.nameConfirmation.pwLocator.fill(currentUserWorkspaceName);
+    await deleteModal.footer.submit.pwLocator.click();
+    await deleteModal.pwLocator.waitFor({ state: 'hidden' });
+  });
+
+  test('View Toggle', async ({ authedPage }) => {
+    const workspaceList = new WorkspaceList(authedPage);
+
+    await workspaceList.gridListRadioGroup.list.pwLocator.click();
+
+    const idSortedWorkspaceNames = _.orderBy(workspaces, 'id', 'desc').map((w) => w.name);
+
+    expect(await workspaceList.table.table.rows.nth(0).name.pwLocator.textContent()).toEqual(
+      idSortedWorkspaceNames[0],
+    );
+
+    // cleanup:
+    await workspaceList.gridListRadioGroup.grid.pwLocator.click();
```

Comment:
can you move this to beforeEach?

Reply:
might be best to make a module level beforeEach for this and that sort beforeEach thing we added

Reply:
actually I realized that this is only needed to support the `beforeEach` in `With a Workspace`, which I actually don't think is necessary.